lib/sparkql/parser_tools.rb

```diff
@@ -42,12 +42,31 @@ def tokenize_conjunction(exp1, conj, exp2)
     exp1 + exp2
   end
 
-  def tokenize_unary_conjunction(conj, exp)
-    exp.first[:unary] = conj
+  def tokenize_unary_conjunction(exp1, conj, exp2)
+    #Unary conjuncion 'Not' is 'And Not'
+
+    exp2.first[:conjunction] = 'And'
+    exp2.first[:conjunction_level] = @lexer.level
+
+    flip_expression_if_necessary(exp2, conj)
+
+    exp1 + exp2
+  end
+
+  def tokenize_unary(conj, exp)
+    flip_expression_if_necessary(exp, conj)
     exp.first[:unary_level] = @lexer.level
```

Comment:
This sounds right... only use the unary_level that matched the first not. However, this willl stick around even when we flip the not off. I don't think it's used in any case.